server/store/peer_store.go

```diff
@@ -24,6 +24,12 @@ type PeerStore interface {
 	GraduateLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
 	AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWant int) (peers, peers6 []chihaya.Peer, err error)
 	CollectGarbage(cutoff time.Time) error
+
+	GetSeeders(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
+	GetLeechers(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
+
+	NumSeeders(infoHash chihaya.InfoHash) (uint32, error)
+	NumLeechers(infoHash chihaya.InfoHash) (uint32, error)
```

Comment:
I had `int` in mind because it's so much easier to handle without having to convert back and forth between `uint` and `int`, plus the length of a map will always fit into an int afaik